src/components/TimeAndSalary/common/injectHideContentBlock.js

```diff
@@ -6,7 +6,7 @@ import cn from 'classnames';
 export default hideRange => rows => {
   const hideIndex = hideRange[0];
   const nHides = hideRange.length;
-
+  const shareLink = '/share';
```

Comment:
這邊可以用
```suggestion
const shareLink = useShareLink();
```

使用時上面要 `import { useShareLink } from 'hooks/experiments';`

Reply:
了解，若使用該路徑，跳出的對話框不會隱藏，但後面的確已跳轉，這部分我再研究看看css
![image](https://user-images.githubusercontent.com/32699556/128487044-5ed4ba81-bc8a-45d3-b40f-cf6e8d4ecfab.png)

Reply:
@jason1004156  剛稍微看了一下，關閉對話框的函式為`onAuthenticatedClick`，但在code在最後送進`<AuthenticatedButton...>`的時候餵錯了，變成bug。修復方式是
```js
// CallToLoginShareButton.js

<AuthenticatedButton to={to} onAuthenticatedClick={onAuthenticatedClick}> // 改成 onClick={onAuthenticatedClick}
  {isLoginText}
</AuthenticatedButton>
```
以及
```js
const AuthenticatedButton = ({ to, onAuthenticatedClick, children }) => ( // 改為 onClick
  <Link
    className={cn('buttonCircleM', 'buttonBlack2')}
    to={to}
    onClick={onAuthenticatedClick} // 改為 onClick
  >
    {children}
  </Link>
);

AuthenticatedButton.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  onAuthenticatedClick: PropTypes.func, // 改為 onClick
};
```
就可以在跳轉之餘也關閉對話框了 :) 謝謝你發現這個問題！

Reply:
@peteranny 非常感謝!! 我還在摸索Reat的功能XD
我今天要驗證時發現搜尋都回傳ERROR500耶，請問SERVER發生什麼事情了嗎?

Reply:
Local Develop 用的 API 有修復了~